contracts/test/integration/IntegrationTestMakerRouter.t.sol

```diff
@@ -0,0 +1,118 @@
+
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+
+    uint256 public constant mintAmount = 200_000_000e18;
+
+    function setUp() public {
+        makerRouter = new MakerRouter(
+            MainnetAddresses.CORE,
+            daiPSM,
+            feiPSM,
+            dai,
+            fei
+        );
+
+        fei.approve(address(makerRouter), type(uint256).max);
+        dai.approve(address(makerRouter), type(uint256).max);
+
+        vm.prank(MainnetAddresses.DAI_USDC_USDT_CURVE_POOL);
+        dai.transfer(address(feiPSM), mintAmount + 100_000_000e18);
+
+        vm.prank(MainnetAddresses.FEI_GOVERNOR);
+        fei.mint(address(this), mintAmount);
+
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantGovernor(address(this));
+    }
+
+    function testSwapFeiForDai(uint64 amountFeiIn) public {
+        vm.assume(amountFeiIn > 1e18);
+
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(amountFeiIn);
+        makerRouter.swapFeiForDai(amountFeiIn, minDaiAmountOut, address(this));
+
+        assertEq(minDaiAmountOut, dai.balanceOf(address(this)));
+    }
+
+    function testSwapFeiForUsdc(uint64 amountFeiIn) public {
+        vm.assume(amountFeiIn > 1e18);
+
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(amountFeiIn);
+        makerRouter.swapFeiForUsdc(amountFeiIn, minDaiAmountOut, address(this));
+
+        assertEq((minDaiAmountOut / 1e12), usdc.balanceOf(address(this)));
+    }
+
+    function testSwapFeiForUsdcAndDai(uint64 amountFeiIn, uint8 ratioUSDC)
+        public
+    {
+        vm.assume(amountFeiIn > 1e18);
+
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(amountFeiIn);
+        makerRouter.swapFeiForUsdcAndDai(
+            amountFeiIn,
+            minDaiAmountOut,
+            ratioUSDC,
+            address(this)
+        );
+
+        uint256 usdcAmount = (minDaiAmountOut * ratioUSDC) / 10000;
+
+        assertEq((minDaiAmountOut - usdcAmount), dai.balanceOf(address(this)));
+        assertEq((usdcAmount / 1e12), usdc.balanceOf(address(this)));
+    }
+
+    function testSwapAllFeiForDai() public {
+        makerRouter.swapAllFeiForDai(address(this));
+
+        assertEq(
+            dai.balanceOf(address(this)),
+            feiPSM.getRedeemAmountOut(mintAmount)
+        );
+    }
+
+    function testSwapAllFeiForUsdc() public {
+        makerRouter.swapAllFeiForUsdc(address(this));
+
+        assertEq(
+            usdc.balanceOf(address(this)),
+            feiPSM.getRedeemAmountOut(mintAmount) / 1e12
+        );
+    }
+
+    function testSwapAllFeiForUsdcAndDai(uint8 ratioUSDC) public {
+        uint256 minDaiAmountOut = feiPSM.getRedeemAmountOut(mintAmount);
+
+        makerRouter.swapAllFeiForUsdcAndDai(address(this), ratioUSDC);
+        uint256 usdcAmount = (minDaiAmountOut * ratioUSDC) / 10000;
+
+        assertEq(usdc.balanceOf(address(this)), usdcAmount / 1e12);
+        assertEq(dai.balanceOf(address(this)), minDaiAmountOut - usdcAmount);
+    }
```

Comment:
add tests here that only the governor or PCV controller can call the router functions

Reply:
just add a test for each function that calls as a non governor or pcv controller where things revert.

also add a test where this test smart contract is a pcv controller and assert that works as well